homeassistant/components/light/everlights.py

```diff
@@ -0,0 +1,182 @@
+from typing import Tuple
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOSTS
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_HS_COLOR, ATTR_EFFECT,
+    SUPPORT_BRIGHTNESS, SUPPORT_EFFECT, SUPPORT_COLOR,
+    Light, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.color as color_util
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.exceptions import PlatformNotReady
+
+REQUIREMENTS = ['pyeverlights==0.1.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'everlights'
+
+SUPPORT_EVERLIGHTS = (SUPPORT_EFFECT | SUPPORT_BRIGHTNESS | SUPPORT_COLOR)
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOSTS, default=[]): [cv.string],
+})
+
+
+def color_rgb_to_int(red: int, green: int, blue: int) -> int:
+    """Return a RGB color as an integer."""
+    return red*256*256+green*256+blue
+
+
+def color_int_to_rgb(value: int) -> Tuple[int, int, int]:
+    """Return an RGB tuple from an integer."""
+    return (value >> 16, (value >> 8) & 0xff, value & 0xff)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the EverLights lights from configuration.yaml."""
+    import pyeverlights
+    lights = []
+
+    for ipaddr in config[CONF_HOSTS]:
+        api = pyeverlights.EverLights(ipaddr,
+                                      async_get_clientsession(hass))
+
+        try:
+            status = await api.get_status()
+
+            effects = await api.get_all_patterns()
+
+        except pyeverlights.ConnectionError:
+            raise PlatformNotReady
+
+        else:
+            lights.append(EverLightsLight(api, pyeverlights.ZONE_1,
+                                          status, effects))
+            lights.append(EverLightsLight(api, pyeverlights.ZONE_2,
+                                          status, effects))
+
+    async_add_entities(lights)
+
+
+class EverLightsLight(Light):
+    """Representation of a Flux light."""
+
+    def __init__(self, api, channel, status, effects):
+        """Initialize the light."""
+        self._api = api
+        self._channel = channel
+        self._status = status
+        self._effects = effects
+        self._mac = status['mac']
+        self._error_reported = False
+        self._hs_color = [255, 255]
+        self._brightness = 255
+        self._effect = None
+        self._available = True
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._mac+'-'+str(self._channel)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return 'EverLights '+self._mac+' Zone '+str(self._channel)
```

Comment:
Why don't you use:

EVERLIGHTS_NAME_FORMAT = 'Everlights {} Zone {}'

EVERLIGHTS_NAME_FORMAT.format(self._mac, self._channel)